bundle/metadata/metadata.go

```diff
@@ -7,11 +7,13 @@ import (
 const Version = 1
 
 type Bundle struct {
-	Git config.Git `json:"git,omitempty"`
+	Git          config.Git `json:"git,omitempty"`
+	SourceLinked bool       `json:"source_linked"`
```

Comment:
Regarding this and the `git_folder_path` properties, the structure of the metadata file matches 1:1 with the bundle configuration schema. This no longer holds with this PR. For derived properties such as these, I propose adding a new top level struct that doesn't map to the configuration to isolate them.

It can be called `metadata`, for example. Open to other suggestions.

@shreyas-goenka Ideas for names to make the distinction clear?